common/persistence/pinot/pinot_visibility_store.go

```diff
@@ -720,20 +720,19 @@ func (s *PinotQuerySearchField) resetSearchField() {
 
 func (s *PinotQuerySearchField) addEqual(obj string, val interface{}) {
 	s.checkFirstSearchField()
-	if _, ok := val.(string); ok {
-		val = fmt.Sprintf("'%s'", val)
-	} else {
-		val = fmt.Sprintf("%v", val)
+	switch val.(type) {
+	case string:
+		s.string += fmt.Sprintf("%s = '%s'\n", obj, val)
+	case int32:
+		s.string += fmt.Sprintf("%s = %d\n", obj, val)
+	default:
+		s.string += fmt.Sprintf("%s = %v\n", obj, val)
 	}
-
-	quotedVal := fmt.Sprintf("%s", val)
-	s.string += fmt.Sprintf("%s = %s\n", obj, quotedVal)
 }
 
 func (s *PinotQuerySearchField) addMatch(obj string, val interface{}) {
 	s.checkFirstSearchField()
-
-	s.string += fmt.Sprintf("text_match(%s, '\"%s\"')\n", obj, val)
+	s.string += fmt.Sprintf("REGEXP_LIKE(%s, '^.*%s.*$')\n", obj, val)
```

Comment:
Once we add the text index, we should use Text_Match for partial match.
Btw, for regexp like query, partial match can timeout for large cluster, it can work with prefix match which is REGEXP_LIKE(%s, '^%s.*$')

Reply:
We can update and test after Pinot team enable text index for us.